Documentation/ceph-csi-troubleshooting.md

```diff
@@ -441,3 +441,13 @@ $ rbd ls --id=csi-rbd-node -m=10.111.136.166:6789 --key=AQDpIQhg+v83EhAAgLboWIbl
 ```
 
 Where `-m` is one of the mon endpoints and the `--key` is the key used by the CSI driver for accessing the Ceph cluster.
+
+## Parallel RBD PVC creation hangs for new BlockPool
+
+This issue is specifically present in CephCSI `v3.4.x`, used by rook `>=v1.7.1` and occurs when multiple parallel PVCs creation requests are issued on a newly created uninitialized blockpool. Follow the steps below to workaround the issue:
```

Comment:
Do we currently have a CSI release with the fix? I'm still not clear in what version of CSI this is or will be fixed.

Reply:
> Do we currently have a CSI release with the fix? 

@travisn
No, we need ceph pacific release with fix https://github.com/ceph/ceph/pull/43113 to build a new cephcsi built on ceph pacific  to resolve this issue.


>I'm still not clear in what version of CSI this is or will be fixed. 

This will be an issue in CSI `v3.4.0` (and also `v3.4.1` which will come out soon).

From https://github.com/ceph/ceph/pull/43113#issuecomment-923918074 ,
We should be able to pick up the fix with next ceph pacific release in cephcsi `v3.4.2` and make changes in the WA doc at that time.

Reply:
@travisn @Rakshith-R instead of documenting how about new cephcsi 3.4.1 with octopus as the base image https://github.com/ceph/ceph-csi/issues/2521#issuecomment-928809267. If we do that we don't need to document any workaround.

Reply:
> @travisn @Rakshith-R instead of documenting how about new cephcsi 3.4.1 with octopus as the base image [ceph/ceph-csi#2521 (comment)](https://github.com/ceph/ceph-csi/issues/2521#issuecomment-928809267). If we do that we don't need to document any workaround.


This is not an option since with ceph octopus `deep_copy()` does not work as expected, refer : https://github.com/ceph/ceph-csi/issues/2521#issuecomment-928825758

Or rook can call `rbd pool init <pool_name>` right after creation?
(IMO since it will be used for rbd images so no harm to initialize it too ?)

@idryomov @travisn Do you think its a more preferrable solution ? 

cc  @Madhu-1

Reply:
Sounds better instead of documentation :+1:

Reply:
So we can avoid this issue if Rook always calls `rbd pool init <pool_name>` immediately after pool creation?

Reply:
> So we can avoid this issue if Rook always calls `rbd pool init <pool_name>` immediately after pool creation? And we would just do this upon creation of CephBlockPool CRs?

Yes, it just needs to be called once. If it is done right after pool create/ setting property, it will avoid this issue.

https://github.com/rook/rook/issues/8696#issuecomment-918038849

Reply:
> > So we can avoid this issue if Rook always calls `rbd pool init <pool_name>` immediately after pool creation? And we would just do this upon creation of CephBlockPool CRs?
> 
> Yes, it just needs to be called once. If it is done right after pool create/ setting property, it will avoid this issue.
> 
> [#8696 (comment)](https://github.com/rook/rook/issues/8696#issuecomment-918038849)

@Rakshith-R Are you planning on this change in Rook, or somebody else should pick it up?